habitat_baselines/evaluate_ppo.py

```diff
@@ -5,20 +5,59 @@
 # LICENSE file in the root directory of this source tree.
 
 import argparse
-
+import cv2
+import numpy as np
+import os
 import torch
-
+import time
 import habitat
+from habitat import logger
+from torch.utils import tensorboard
+from habitat.utils.visualizations.utils import images_to_video
 from config.default import get_config as cfg_baseline
 from habitat.config.default import get_config
 from rl.ppo import PPO, Policy
 from rl.ppo.utils import batch_obs
 from train_ppo import make_env_fn
+from habitat.utils.visualizations import maps
+
+
+def poll_checkpoint_folder(checkpoint_folder, previous_ckpt_ind):
+    assert os.path.isdir(checkpoint_folder), "invalid checkpoint folder path"
+    models = os.listdir(checkpoint_folder)
+    models.sort(key=lambda x: int(x.strip().split(".")[1]))
+    ind = previous_ckpt_ind + 1
+    if ind < len(models):
+        return os.path.join(checkpoint_folder, models[ind])
+    return None
+
+
+def images_to_tb_video(video_name, step_idx, frames, writer, fps=10):
+    frame_tensors = [
+        torch.from_numpy(np_arr).unsqueeze(0) for np_arr in frames
+    ]
+    video_tensor = torch.cat(tuple(frame_tensors))
+    video_tensor = video_tensor.permute(0, 3, 1, 2).unsqueeze(
+        0
+    )  # Shape (N,T,C,H,W)
+    writer.add_video(video_name, video_tensor, fps=fps, global_step=step_idx)
+
+
+def draw_collision(view):
+    size = view.shape[0]
+    strip_width = int(size / 20)
+    mask = np.ones((size, size))
+    mask[strip_width:-strip_width, strip_width:-strip_width] = 0
+    mask = mask == 1
+    alpha = 0.4
+    view[mask] = (alpha * np.array([255, 0, 0]) + (1.0 - alpha) * view)[mask]
```

Comment:
You might want to look into `cv2.addWeighted` function which provides even more granular control.